### ISSUE TITLE
ComplexMethod should treat "return when" as "single when" expression when ignoring

### ISSUE THREAD

**Issue by dimsuz** [arturbosch/detekt#1054]
## Expected Behavior

When detekt is configured like this:

```
  ComplexMethod:
    active: true
    threshold: 10
    ignoreSingleWhenExpression: true
```
This kind of `when` should be treated as "single when expression" and ignored
```kotlin
fun someFunction(s: String): Boolean {
  return when (s) {
     "bear" -> true
     "log" -> false
     // ... 15 more branches
  }
}
```

## Current Behavior
Currently the code above will report a `ComplexMethod` rule failure, because it ignores only "non-expression "when's", like

```kotlin
fun someFunction(s: String): Unit {
  when (s) {
     "bear" -> doSomething()
     "log" -> doSomethingElse()
     // ... 15 more branches
  }
}
```

If you are OK with the proposed change I could create a PR.

**Comment by schalkms** [comment]
This shouldn't be the case.
That would be nice if you could create a PR.
